tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupTargetViewModel.cs

```diff
@@ -395,63 +397,65 @@ public async Task UpdateListViewModelList(ComputeSystemsLoadedData data)
-        foreach (var mapping in data.DevIdToComputeSystemMap.Where(kvp =>
-            kvp.Value.Result.Status == ProviderOperationStatus.Failure))
+        var computeSystemListViewModels = new List<ComputeSystemsListViewModel>();
+        var allComputeSystems = new List<ComputeSystemCache>();
+        foreach (var devIdToComputeSystemResultPair in data.DevIdToComputeSystemMap)
         {
-            data.DevIdToComputeSystemMap.Remove(mapping.Key);
-        }
+            // Remove the mappings that failed to load.
+            // The errors are already handled by the notification helper.
+            if (devIdToComputeSystemResultPair.Value.Result.Status == ProviderOperationStatus.Failure)
+            {
+                continue;
+            }
+
+            var listViewModel = new ComputeSystemsListViewModel(data.ProviderDetails, devIdToComputeSystemResultPair);
 
-        var curListViewModel = new ComputeSystemsListViewModel(data);
+            if (listViewModel.ComputeSystems.Count > 0)
+            {
+                computeSystemListViewModels.Add(listViewModel);
+                allComputeSystems.AddRange(listViewModel.ComputeSystems);
+            }
+        }
 
         // Fetch data for all compute systems that support the ApplyConfiguration flag in parallel
         // on thread pool to avoid calling expensive OOP operations on the UI thread.
-        await Parallel.ForEachAsync(curListViewModel.ComputeSystems, async (computeSystem, token) =>
+        await Parallel.ForEachAsync(allComputeSystems, async (computeSystem, token) =>
         {
-            if (computeSystem.SupportedOperations.Value.HasFlag(ComputeSystemOperations.ApplyConfiguration))
-            {
-                await computeSystem.FetchDataAsync();
-            }
+            await computeSystem.FetchDataAsync();
         });
 
         await _dispatcherQueue.EnqueueAsync(async () =>
         {
-            foreach (var computeSystem in curListViewModel.ComputeSystems)
+            foreach (var listViewModel in computeSystemListViewModels)
             {
-                // Remove any cards that don't support the ApplyConfiguration flag.
-                if (!computeSystem.SupportedOperations.Value.HasFlag(ComputeSystemOperations.ApplyConfiguration))
+                foreach (var computeSystem in listViewModel.ComputeSystems)
                 {
-                    continue;
+                    var packageFullName = data.ProviderDetails.ExtensionWrapper.PackageFullName;
+                    var card = await _computeSystemViewModelFactory.CreateCardViewModelAsync(
+                        ComputeSystemManagerObj,
+                        computeSystem,
+                        data.ProviderDetails.ComputeSystemProvider,
+                        packageFullName,
+                        _dispatcherQueue);
+
+                    // Don't show environments that aren't in a state to configure
+                    if (!ShouldShowCard(card.CardState))

```

Comment:
The card needs to be made before you can check if the environment is in the correct state?

Reply:
For this yes, for example if the compute system was in the creation state we don't want to show it in the UI but in the future we wanted to raise an event that would then make this appear in the UI when its done. Currently we don't have an event but it was more of a future proofing thing from a previous change.

Reply:
I see.  You _could_ ask the compute system what state it is in.  But, because the compute system could be ready when DevHome is open, this saves the compute system and the card in memory to show it if neeeded.